src/extension_commands.mli

```diff
@@ -1,8 +1,11 @@
 type command = private
   { id : string
-  ; handler : Extension_instance.t -> unit -> unit
+  ; handler : Extension_instance.t -> args:Ojs.t list -> unit
   }
 
+val make_command :
+  string -> (Extension_instance.t -> args:Ojs.t list -> unit) -> command
+
 (** Module to manage commands[1] across the extension.
 
     Module does not have public API for command creation on purpose. One should
```

Comment:
The comment says specifically that there is no `make_command` in the module API, so that the module handles command disposable subscription across the whole extension. 

It may be worth to either respect the module's goal of handling all commands (e.g., by exposing current unexposed `command` and use it along with `register_all_commands`) or remove that comment.

Reply:
This one was a bit awkward, I wanted to respect the design decision for this module, but the `refresh-switch` command needs access to the event emitter created in the treeview. Passing it in the `register_all_commands` did not make sense IMO.

That's a problem I figure will happen often, especially now that we have a UI that will need the creation of more commands. With this in mind, I think it might make sense to revisit the design of the extension commands to have a single pattern (either we create all of the commands in `vscode_ocaml_platform.ml`, because they have dependencies that should be available here, or we keep them in this module, but it will not be properly isolated).

Reply:
You decide how to go forward, but if you want to keep the current design: 

Expose `command : name -> handler -> ()` from `extension_commands.ml` that creates a command and adds it to the module-local mutable list `commands`. Call `register_all_commands` in `vscode_ocaml_platform.ml`, which uses `commands` to register all commands with the ext context.